index.html

```diff
@@ -0,0 +1,370 @@
+        };
+
+      var tbody = d3.select("tbody");
+
+      function updateTable() {
+        var data = states.slice();
+                     
+          // Erase any existing html code within tbody by replacing it with a blank string "".
+        tbody.html("");
+        // Binding data - this will be a very common task in our use of D3. Start by making a selectAll of <tr> elements that do not yet exist (they will soon!).
+        // Then we bind our data (schools) with .data and .enter()
+        // Lastly, we append the <tr> elements, which are become our current selection because of the prior selectAll("tr")
+      
+
+
+        if(options.filtered) {
+          if (options.filtered ==='no rest') {
+            data=data.filter(function (d) {return d.Restrictions==='No Restrictions'; });
+          } else if (options.filtered ==='prison') {
+            data=data.filter(function (d) {return d.Restrictions==='Restrictions in Prison'; });
+          } else if (options.filtered ==='parole') {
+            data=data.filter(function (d) {return d.Restrictions==='Restrictions While on Parole'; });
+          } else if (options.filtered ==='probation') {
+            data=data.filter(function (d) {return d.Restrictions==='Restrictions While on Probation'; });
+          } else if (options.filtered ==='post') {
+            data=data.filter(function (d) {return d.Restrictions==='Restrictions Post-Sentence'; });
+          }
+        }
+
+          var rows = tbody.selectAll("tr")
+          .data(data)
+          .enter()
+          .append("tr");
+          console.log(rows);
+        // If you run, console.log(rows), you should now see an array of <tr> elements printed out to the console, each of which should contain a __data__ property containing one object from our JSON array.
+        // console.log(rows);
+        // Take the rows variable and append a <td> elements containing d.name - the name of each schools for each bound data element.
+        rows.append("td")
+          .html(function(d) { return d.State });
+        // Same as above, except for level of the school.
+        rows.append("td")
+          .html(function(d) { return d.StateAbbr });
+        // Same as above, except for the number of students enrolled in the school.
+        rows.append("td")
+          .html(function(d) { return d.Restrictions });
+
+        rows.append("td")
+          .html(function(d) {  return d.felon + ' %' });
+        // Same as above, calculating a percent of the space allowed that the div should fill:
+        rows.append("td")
+          .append("div")
+          .classed("bar", true)
+          .style("width", function(d) { return ((d.felon / MAX_DIS) * 100) + '%' });
+
+
+
+        
+        };
+
+      updateTable();
+
+    d3.select("#sort-asc")
+      .on("click", function(){
+      d3.selectAll("tbody tr") 
+      .sort(function(a, b) {
+                return d3.ascending(a.felon, b.felon);
+                            });
```

Comment:
Note that this gets overwritten every time we change the filter. This works, but we might run into issues eventually (imagine, for example, if we had an active state on these sorts; now you have to remember to wipe that active state every time you perform an unrelated action — changing the filter). This is the advantage of doing it every time inside updateTable() — we can ensure consistency between what we think are our options (in the `options` object) and what is shown in the browser.

Props for using selection.sort() correctly, though.